dandi/utils.py

```diff
@@ -890,3 +891,66 @@ def joinurl(base: str, path: str) -> str:
         return path
     else:
         return base.rstrip("/") + "/" + path.lstrip("/")
+
+
+def get_retry_after(response: requests.Response) -> Optional[int]:
+    """If provided and parsed ok, returns duration in seconds to sleep before retry.
+
+    If not provided in the response header `Retry-After`, would
+    return None.
+    If parsing fails, or provided date/sleep does not make sense
+    since either too far in the past (over 2 seconds) or in the future
+    (over a week), would return None.
+    """
+    if_unparsable = None
+    retry_after = response.headers.get("Retry-After")
+    if retry_after is None:
+        return None
+    sleep_amount: int | None
+    if retry_after.isdecimal():
+        sleep_amount = int(retry_after)
+    else:
```

Comment:
```suggestion
    try:
        sleep_amount = int(retry_after)
    except ValueError:
```

Reply:
any benefit?  I could potentially see how it would be more robust but not really "pragmatically"

by doing in `else:` we would avoid potential "while handling exception ... another exception" spaghetti of exceptions.

Reply:
The purpose of the code is to parse `retry_after` as an integer if possible, and you can do that just with `try: sleep_amount = int(retry_after); except ...`.  Doing a "look before you leap" test with `isdecimal()` wastes CPU cycles and doesn't even exactly align with the set of strings accepted by `int()` (e.g., the latter accepts leading whitespace and negative numbers).

It's just better, more Pythonic code.